Versioning in WordPress/Rebasing Intro.md

```diff
@@ -0,0 +1,75 @@
+The following is a quick rebasing how to and when tutorial.
+
+__Scenario__: You've made a feature (`feat1`) and you need this feature for some other feature (`feat2`), so that you can continue your work. You can create a new branch from the feature branch and continue developing. During your development time, you've made a PR for the original feature.
+The PR has been approved and you merge it to the `staging` branch.
+
+Now you can rebase the new feature to the `staging` branch.
+
+While on the feature branch (`feat2`) commit all that you've developed so far. Check out the `staging` branch
+
+```bash
+git checkout staging
+```
+
+Pull all the changes to your local `staging` branch.
+
+```bash
+git pull origin staging
+```
+
+This will depend on your git settings (you might not need to define the `origin` part).
+
+After that, checkout the `feat2` branch
+
+```bash
+git checkout feat2
+```
+
+Now you can do an interactive rebase
+
+```bash
+git rebase -i origin/staging
+```
+
+You should see a commit message in the editor that you've set in the `.gitconfig` settings.
+
+You can quit (in nano it's cmd x, and in vim it's ctrl q if I'm not mistaken).
+
+Now the interactive rebase process has begun. If you have a conflict it will tell you which file is in the conflicted state. Fix the conflict and write
+
+```bash
+git add .
+```
+
+This is one of the rare cases you'll want to use the git add all command. Usually, __NEVER__ add all in one commit unless you have a small change on one feature. Otherwise, you are losing the granularity and breaking the single responsibility principle (SRP) of your commits. Remember, commits are telling a story. You should be able to read the commit history and understand what the developer was trying to do.
+
+⚠️ __DO NOT COMMIT!!!!!!__ So that means __don't type__ `git commit` command.
+
+When you've fixed all the conflicts and added the fixed files, type
+
+```bash
+git rebase --continue
+```
+
+You may need to fix some more conflicts, but maybe you won't. 
+If all is ok, the rebase will be successful, and you can push your branch with
+
+```bash
+git push --force
+```
+
+You need to use force push because you've changed the git history because the hashes of the git commits have changed.
+
+You can also use
+
+```bash
+git push --force-with-lease origin
+```
+
+which is a more secure way to avoid overwriting in case somebody else worked on the remote branch you've just rebased.
+
+That shouldn't be problematic because the good __rule of thumb__ is: If two people are working on a single feature branch, at the same time, don't rebase. Ever. You're going to change the commit hashes and the other person will get tons of conflicts or errors, and then they'll try to rebase, and you'll chase each other in circles and curse the day you did rebase 😅.
```

Comment:
```suggestion
That shouldn't be problematic because the good __rule of thumb__ is: If two people are working on a single feature branch (which you should never do, each person should work in it's own branch), at the same time, don't rebase. Ever. You're going to change the commit hashes and the other person will get tons of conflicts or errors, and then they'll try to rebase, and you'll chase each other in circles and curse the day you did rebase 😅.
```